fedimint-client/src/module/init.rs

```diff
@@ -102,16 +153,16 @@ where
     ) -> anyhow::Result<DynClientModule> {
         let typed_cfg: &<<T as fedimint_core::module::ExtendsCommonModuleInit>::Common as CommonModuleInit>::ClientConfig = cfg.cast()?;
         Ok(self
-            .init(
+            .init(&ClientModuleInitArgs {
```

Comment:
One thing I remembered when reviewing the server version of this PR: wouldn't we want the same struct as a dyn version? I guess it's less necssary since we view the dyn stuff as internal API, not to be touched by module authors?

Reply:
I removed the PR from the merge queue for now, just merge if that was your reasoning and not an oversight.

Reply:
Crossed my mind, but as it is an internal signature it doesn't seem necessary. It has only one generic implementation, so it doesn't seem to gain anthing, other than maybe keeping it more 1:1 to the typed version. I mean - it would be like taking any internal function with N arguments and making it a struct. 

@elsirion .